### DISCUSSION TITLE
Draw a line chart in a time series

### DISCUSSION THREAD

**Discussion by plsmarann** [patrykandpatrick/vico#793] [Questions]
### Question

Draw a line chart on a time series where time series > y value , there are x  values from 1pm to 10 pm and the non value y values start form 2pm to 6 pm need the point to start and end in the above mentioned time and not have the line dip to 0 

### Vico version(s)

1.14.0

### UI framework(s)

Jetpack Compose

**Comment by joseandroidengineer** [comment]
Just shrink your range from 1 -> 10 to 2 -> 6. Why show x - values that have no y - values?

**Reply by plsmarann** [reply]
I need to have a chart that looks something like this 
<img width="512" alt="Screenshot 2024-07-17 at 10 43 07 AM" src="https://github.com/user-attachments/assets/58b857f6-72fc-4e35-9442-1c8bfe8ba688">

**Reply by patrickmichalik** [reply]
Hello! It looks like there was a problem with the screenshot upload, @plsmarann. Could you try uploading the screenshot again? I believe I understand what you’re referring to, but a picture for confirmation would be helpful.

**Reply by plsmarann** [reply]
@patrickmichalik  have updated the image

**Reply by plsmarann** [reply]
@patrickmichalik  any update here

**Reply by patrickmichalik** [reply]
Thanks, @plsmarann! You can use `AxisValuesOverrider` to achieve this. For a static _x_ range, you can use the following:

```kt
lineChart(
  axisValuesOverrider = remember { AxisValuesOverrider.fixed(minX = 1f, maxX = 10f) },
  /* ... */
)
```

If you need the _x_ range to change over time, you should instead create an `AxisValuesOverrider` implementation with dynamic behavior. Let me know if you have any questions.

For completeness, I’d like to add that in Vico 2, `AxisValuesOverrider` is called `AxisValueOverrider`. It has [a dedicated wiki section](https://patrykandpatrick.com/vico/wiki/cartesian-charts/layers/#axisvalueoverrider).

**Reply by plsmarann** [reply]
![Screenshot_20240806-172109](https://github.com/user-attachments/assets/126cb0d6-3994-4cf5-aac2-d094494f9a2e)
 i need the line to start from around 90 and end there , right now the value dips down for the end value

**Reply by Gowsky** [reply]
@plsmarann just like I [said here](https://github.com/patrykandpatrick/vico/discussions/759#discussioncomment-9640264), this suggests that you're adding unwanted _y_ values.

The series should only include the _y_ values that you want to see on the line. You can extend the _x_ axis range by using `AxisValueOverrider` as @patrickmichalik mentioned.

The line would then not dip.
![image](https://github.com/user-attachments/assets/e76a0356-a7fe-407a-9138-c8862fe74b7e)

**Reply by plsmarann** [reply]
how can i create a Broken-line graphs like 
![download](https://github.com/user-attachments/assets/f623ac8d-c5c3-4b22-84d3-a6f5de5ed581)

**Reply by Gowsky** [reply]
@plsmarann you have to add each line segment as a separate series.
